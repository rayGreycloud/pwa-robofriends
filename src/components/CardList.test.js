import React from 'react';
import { shallow } from 'enzyme';

import CardList from './CardList';

describe('<CardList />', () => {
  it('should render component', () => {
    const mockRobots = [
      {
        id: 1,
        name: 'Kryten',
        username: 'Wildman',
        email: 'robogod@reddwarf.com',
      },
    ];

    expect(shallow(<CardList robots={mockRobots} />)).toMatchSnapshot();
  });
});
