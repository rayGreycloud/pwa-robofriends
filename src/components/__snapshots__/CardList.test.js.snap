// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardList /> should render component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardList
    robots={
        Array [
            Object {
              "email": "robogod@reddwarf.com",
              "id": 1,
              "name": "Kryten",
              "username": "Wildman",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          email="robogod@reddwarf.com"
          id={1}
          name="Kryten"
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "email": "robogod@reddwarf.com",
          "id": 1,
          "name": "Kryten",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            email="robogod@reddwarf.com"
            id={1}
            name="Kryten"
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "email": "robogod@reddwarf.com",
            "id": 1,
            "name": "Kryten",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
